Abracadabra/abracadabra_web_app/src/components/SubjectQuestionAnswer.js

```diff
@@ -1,41 +1,164 @@
 import { Card, Row, Col } from "react-bootstrap";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, onClick } from "react";
 import {
   faChevronUp,
   faChevronDown,
   faFlag,
   faCheck,
 } from "@fortawesome/free-solid-svg-icons";
+import VotesService from "../services/VotesService"
+import QuestionService from "../services/QuestionService"
+import AnswerService from "../services/AnswerService"
 
 export default function Answer({ answer }) {
-  const [voted, setVoted] = useState();
+  const [date, setDate] = useState();
+  const [isloggedin, setIsLoggedIn] = useState(false);
+  const [totalvotes, setTotalVotes] = useState(answer.upvotes - answer.downvotes)
+  const [voted, setVoted] = useState(false)
+  const [rendered, setRendered] = useState(false)
   const [vote, setVote] = useState({
-    questionid: "",
-    vote: "",
-  });
-  const handleClick = (amount) => {
-    setVote({ ...comment, vote: amount });
-  };
+    AnswerId: answer.id,
+    vote: ""
+  })
+  
+  const UpdateVotesAnswers = () => {
+    AnswerService.GetAnswer(answer.id).then((res) => {
+      console.log(res);
+      console.log(res.data);
+      setTotalVotes(res.data.upvotes - res.data.downvotes)
+    })
+      .catch(() => {
+      });
+  }
 
-  const handlePost = () => {
-    VotesService.PostVoteAnswer(vote)
-      .then((res) => {
+  useEffect(() => {
+    const tokenExist = localStorage.getItem("Token");
+    if (tokenExist) {
+      setIsLoggedIn(true);
+       VotesService.GetAnswerVote(answer.id).then((res) => {
         console.log(res);
         console.log(res.data);
+        if (res.data.vote == 1|-1){
+        setVote({ ...vote, vote: res.data.vote })
+        setVoted(true)
+        }
       })
-      .catch((error) => {
-        console.log(error.response.data);
+        .catch(() => {
```

Comment:
It is bad practise not to do anything when you catch an error. You can display a message or show something to inform the user of the error that occurred.